app/models/location.rb

```diff
@@ -134,6 +148,49 @@ class Location < AbstractModel
     end
   end
 
+  scope :name_includes,
+        ->(place_name) { where(Location[:name].matches("%#{place_name}%")) }
+  scope :in_region,
+        ->(place_name) { where(Location[:name].matches("%#{place_name}")) }
+  scope :in_box, # Use named parameters (n, s, e, w), any order
+        lambda { |**args|
+          # rubocop disable:Lint/SafeNavigationConsistency
+          # false positive and attempted correction cause Ruby syntax error
+          # and I can't seem to disable the cop -- JDC 2022-07-25
+          if args[:s]&.between?(0, 90) && args[:n]&.between?(0, 90) &&
+             args[:w]&.between?(-180, 180) && args[:e]&.between?(-180, 180) &&
+             args[:s] <= args[:n] &&
+             ((args[:w] <= args[:e]) || (args[:w] >= 0 && args[:e] <= 0))
+
+            # leeway for Float rounding
+            # Fixes a bug where Califoria fixture was not in a box
+            # defined by the fixture's north, south, east, west
+            epsilon = 0.00001
+            n = args[:n] + epsilon
+            s = args[:s] - epsilon
+            e = args[:e] + epsilon
+            w = args[:w] - epsilon
+
+            # Does box straddle 180 deg?
+            if args[:e] < args[:w]
+              where(
+                (Location[:south] >= s).and(Location[:north] <= n).
+                # Location[:west] between w & 180 OR between 180 and e
+                and((Location[:west] >= w).or(Location[:west] <= e)).
+                and((Location[:east] >= w).or(Location[:east] <= e))
+              )
+            else
+              where(
+                (Location[:south] >= s).and(Location[:north] <= n).
+                and(Location[:west] >= w).and(Location[:east] <= e).
+                and(Location[:west] <= Location[:east])
+              )
+            end
```

Comment:
@pellaea This bit is probably worth a second set of eyes.
(I struggled with it when considering just the code text. 
Then drew a back-of-the-napkin sketch and it became simple & clear -- I think.)

Reply:
This looks correct for testing whether the location is entirely enclosed within the given box.

I have a too-late kind of question, though...  Consider the case of a location which is 99.9% within the given box...  Oof.  It gets complicated really quickly if you start considering overlap cases.  Earth overlaps all possible boxes 100%!, but clearly shouldn't match.

Bah, what a mess, keep it simple: your "strictly and completely enclosed" is clear, conceptually simple, and is already written and working! :)  If someone wants something else... well, then they can bloody well write it themselves, ha!

Reply:
Thanks for the review and the question about overlaps. 
Overlaps hadn't crossed my mind. I agree with your answer.